assets/js/datasets.js

```diff
@@ -0,0 +1,196 @@
+'use strict';
+
+window.linqs = window.linqs || {};
+window.linqs.datasets = window.linqs.datasets || {};
+window.linqs.utils = window.linqs.utils || {};
+
+window.linqs.datasets.metadata = window.linqs.datasets.metadata || {};
+window.linqs.datasets.pubs = window.linqs.datasets.pubs || {};
+
+window.linqs.datasets.listAuthors = function(pub) {
+    let authors = '';
+
+    for (let i = 0; i < pub['authors'].length; i++) {
+        let text = pub['authors'][i];
+
+        authors += window.linqs.utils.listAuthor(text, i, pub['authors'], false)
+    }
+
+    return authors;
+};
+
+window.linqs.datasets.bibtex = function(pub, id) {
+    let authors = window.linqs.datasets.listAuthors(pub);
+    return window.linqs.utils.bibtex(pub, authors, id);
+};
+
+window.linqs.datasets.size = function(size) {
+    let sizes = ["B", "KB", "MB", "GB", "TB", "PB"];
+
+    let i = 0;
+    while (i < sizes.length && size >= 1024) {
+        size = size / 1024.0;
+        i++;
+    }
+
+    return size.toFixed(2).toString() + " " +sizes[i];
+};
+
+window.linqs.datasets.makeDownloadInfo = function(downloads) {
+    let downloadInfosHTML = '';
+
+    downloads.forEach(function(download, i) {
+        let link = download['href'];
+        let text = download["text"];
+        let statsHTML = "";
+
+        let sizeRaw = download['size'];
+        let md5 = download['md5'];
+
+        if (sizeRaw != null && md5 != null) {
+            let size = window.linqs.datasets.size(sizeRaw);
+
+            statsHTML = `
+                <div class='size'>
+                    (${size})
+                </div>
+                <div class='hash'>
+                    md5: ${md5}
+                </div>
+            `;
+        }
+
+        let downloadInfoHTML = `
+            <div class='link-info'>
+                <a class='download-link' href='${link}'>${text}</a>
+                ${statsHTML}
+            </div>
+        `;
+
+        downloadInfosHTML += downloadInfoHTML;
+    });
+
+    return downloadInfosHTML;
+};
+
+window.linqs.datasets.makeReference = function(reference) {
+    let refLink = '404.html';
+    reference['links'].forEach(function(link, i) {
+        if (link['label'] == 'paper')
```

Comment:
Why don't other labels get a link?

If this is the "links" key, shouldn't every link have an href?

Reply:
The idea was just to take them directly to the paper. Another option is to take them to the paper specific page for the pub.

Reply:
Okay, I see.
There are two issues here.

First, you can't rely on there always being a paper.
What if it is a tutorial that only has slides/video?
e.g.: https://linqs.github.io/linqs-website/publications/#id:augustine-uai18

Second, we want to link to the pubs page for that paper.
That way, they can look at all the information, the citation, and any other resources (slides, poster, etc).